runtime/src/dpe_crypto.rs

```diff
@@ -78,64 +112,193 @@ impl<'a> Crypto for DpeCrypto<'a> {
+            AlgLen::Bit384 => {
+                let mut hasher = self
+                    .hash_initialize(algs)
+                    .map_err(|_| CryptoError::HashError)?;
+                hasher
+                    .update(measurement.bytes())
+                    .map_err(|_| CryptoError::HashError)?;
+                hasher.update(info).map_err(|_| CryptoError::HashError)?;
+                let context = hasher.finish().map_err(|_| CryptoError::HashError)?;
 
-    fn derive_private_key(
-        &mut self,
-        _algs: AlgLen,
-        _cdi: &Self::Cdi,
-        _label: &[u8],
-        _info: &[u8],
-    ) -> Result<Self::PrivKey, CryptoError> {
-        Err(CryptoError::NotImplemented)
+                hmac384_kdf(
+                    self.hmac384,
+                    KeyReadArgs::new(KEY_ID_RT_CDI).into(),
+                    b"derive_cdi",
+                    Some(context.bytes()),
+                    self.trng,
+                    KeyWriteArgs::new(KEY_ID_DPE_CDI, KeyUsage::default().set_hmac_data_en())
+                        .into(),
+                )
+                .map_err(|_| CryptoError::CryptoLibError)?;
+                Ok(KEY_ID_DPE_CDI)
+            }
+        }
     }
 
-    fn derive_ecdsa_pub(
+    fn derive_key_pair(
         &mut self,
-        _algs: AlgLen,
-        _priv_key: &Self::PrivKey,
-    ) -> Result<EcdsaPub, CryptoError> {
-        Err(CryptoError::NotImplemented)
+        algs: AlgLen,
+        cdi: &Self::Cdi,
+        label: &[u8],
+        info: &[u8],
+    ) -> Result<(Self::PrivKey, EcdsaPub), CryptoError> {
+        match algs {
+            AlgLen::Bit256 => Err(CryptoError::Size),
+            AlgLen::Bit384 => {
+                hmac384_kdf(
+                    self.hmac384,
+                    KeyReadArgs::new(*cdi).into(),
+                    label,
+                    Some(info),
+                    self.trng,
+                    KeyWriteArgs::new(KEY_ID_TMP, KeyUsage::default().set_ecc_key_gen_seed_en())
+                        .into(),
+                )
+                .map_err(|_| CryptoError::CryptoLibError)?;
+
+                let pub_key = self
+                    .ecc384
+                    .key_pair(
+                        &Ecc384Seed::Key(KeyReadArgs::new(KEY_ID_TMP)),
+                        &Array4x12::default(),
+                        self.trng,
+                        KeyWriteArgs::new(
+                            KEY_ID_DPE_PRIV_KEY,
+                            KeyUsage::default().set_ecc_private_key_en(),
+                        )
+                        .into(),
+                    )
+                    .map_err(|_| CryptoError::CryptoLibError)?;
+                let pub_key = EcdsaPub {
+                    x: CryptoBuf::new(pub_key.x.as_bytes(), algs).map_err(|_| CryptoError::Size)?,
+                    y: CryptoBuf::new(pub_key.y.as_bytes(), algs).map_err(|_| CryptoError::Size)?,
+                };
```

Comment:
Using as_bytes doesn't work here. The reason is that the hardware consumes little endian words, while DPE is expecting to get the value as big-endian. Take the example X value `0xAABBCCDD_00112233`

When written to Caliptra's registers, this will have the byte representation `0xDDCCBBAA_33221100`. So we need to do endianness conversions between the two types. But as_bytes just taskes the raw byte representation.

Something like this should work (the `From` implementation in `drivers/src/array.rs` does the endianness conversion)

```rust
CryptoBuf::new(&<[u8; AlgLen::Bit384.size()]>::from(pub_key.x), algs).map_err(|_| CryptoError::Size)?
```